### PR TITLE
Dummy and Clock widget improvements

### PR DESCRIPTION
**Pull request by BasvanH**
Signed-off-by: Bastiaan an Haastrecht b.vanhaastrecht@gmail.com
Adds color customization to Clock and Dummy widget.
Adds option to show state description in Dummy widget.

Closes: #1349
Closes: #1293
Closes: #1294

### CODE REVIEW

File: bundles/org.openhab.ui.habpanel/doc/habpanel.md
Comment (ghys) [inline bundles/org.openhab.ui.habpanel/doc/habpanel.md:171]:
```suggestion
The so-called dummy widget (whose name is explained by historical reasons - it evolved from the first developed widget) displays the current state of an item without any interactivity, along with a label and an optional icon. It has a few options to customize its appeareance.
```

**Review by ghys — COMMENTED** [review]
Many thanks, this looks quite good (especially the i18n stuff), a few minor remarks on the docs, but LGTM.

File: bundles/org.openhab.ui.habpanel/doc/habpanel.md
Comment (ghys) [inline bundles/org.openhab.ui.habpanel/doc/habpanel.md:233]: Not sure this addition was necessary to be emphasized here tbh :)

**Review by ghys — APPROVED** [review]
Merging this small change as non disruptive, besides it was positively reviewed a while ago and I was just waiting for feedback on my doc changes which I ended up reverting yesterday.

**Comment by jsiemon** [comment]
The changes made between M7 and RC1 seem to have broken the Dummy Widget when implemented  with HabPanel and used on an iPad 3 or earlier.  IPad 4and later seem unaffected.  I believe many, not all, HabPanel users use older hardware, otherwise they would likely use MainUI instead of HabPanel.  I view this as a regression for a feature implemented after the features were frozen.

**Comment by kaikreuzer** [comment]
@ghys Can you confirm that those changes might have increased the minimal browser requirements?
I leave it to your judgement on whether this is something that we can live with or if there's any need to act.

**Comment by jsiemon** [comment]
Whatever changes were made seems only to have negatively affected the Dummy Widget.  The Clock Widget displays fine on the iPad 3.

File: bundles/org.openhab.ui.habpanel/web/app/widgets/dummy/dummy.widget.js
Comment (ghys) [inline bundles/org.openhab.ui.habpanel/web/app/widgets/dummy/dummy.widget.js:54]: This is probably the line that is causing problems (arrow functions are iOS Safari 10+, `find` is iOS Safari 8+).
The iPad 3 apparently runs on iOS 9.

Reference
https://caniuse.com/arrow-functions
https://caniuse.com/array-find
https://en.wikipedia.org/wiki/IPad_(3rd_generation)#Software

It could probably be chenged to `options.find((function (element) { return element.value === item.state; })` to make it compatible with iOS 8+.

Comment (jsiemon) [reply bundles/org.openhab.ui.habpanel/web/app/widgets/dummy/dummy.widget.js:54]: That would be great if it can be fixed as proposed.  I know this is old hardware, but it works great with HabPanel otherwise and HabPanel is a great legacy  alternative to the new MainUI that does require more modern hardware to run.  Your efforts are greatly appreciated.

Comment (ghys) [reply bundles/org.openhab.ui.habpanel/web/app/widgets/dummy/dummy.widget.js:54]: #1433 reimplements the code with techniques found elsewhere (and it seems to work in iOS 6).

image

Thank you for reporting that late minute bug.

**Pull request merged**
